modules/lib/lib-node/src/main/resources/lib/xp/node.ts

```diff
@@ -416,15 +416,9 @@ export interface Explanation {
     details: Explanation[];
 }
 
-export interface NodeQueryResult {
+export interface NodeQueryResultBase {
```

Comment:
Since `NodeQueryResult` and  `NodeMultiRepoQueryResult` are almost the same, I would suggest to use the following:
```ts
interface NodeQueryResult {
    total: number;
    count: number;
    aggregations?: AggregationsResult;
    suggestions?: {
        [suggestionName: string]: {
            text: string;
            length: number;
            offset: number;
            options: {
                text: string;
                score: number;
                freq?: number; // only for term
            }[];
        }[];
    }[];
    hits: {
        id: string;
        score: number;
        explanation?: Explanation;
        highlight?: HighlightResult;
    }[];
}

export type NodeMultiRepoQueryResult = NodeQueryResult & {
    hits: {
        repoId: string;
        branch: string;
    }[];
};
```